arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Pair.kt

```diff
@@ -13,8 +14,12 @@ public operator fun <A : Comparable<A>, B : Comparable<B>> Pair<A, B>.compareTo(
   else first
 }
 
-public fun <A, B> Pair<A, B>.combine(SA: Semigroup<A>, SB: Semigroup<B>, b: Pair<A, B>): Pair<A, B> {
+public fun <A, B> Pair<A, B>.combine(combineA: (A, A) -> A, combineB: (B, B) -> B, b: Pair<A, B>): Pair<A, B> {
   val (xa, xb) = this
   val (ya, yb) = b
-  return Pair(SA.run { xa.combine(ya) }, SB.run { xb.combine(yb) })
+  return Pair(combineA(xa, ya), combineB(xb, yb))
```

Comment:
Should we deprecate this method too? Not sure what type class this originally came from 🤔
Does it bring extra value to Arrow? The resulting Kotlin code seems almost simpler than the method:

```kotlin
val pairA = Pair(xa, xb)
val pairB =  Pair(xb, yb)
pairB.combine(combineA, combineB, pairB)
```

vs

```kotlin
val (a, b) = Pair(xa, xb)
val (c, d) =  Pair(xb, yb)
Pair(combineA(a, b), combineB(c, d))
```

Reply:
I don't think it hurts having it there, although I agree it may not see much use...

Reply:
Also Agree the Kotlin code looks simpler

Reply:
> I don't think it hurts having it there, although I agree it may not see much use...

I decided to deprecate this method instead of adding a new signature. I'm not even sure where it originally came from 🤔 `BiFunctor` and `Semigroup`?